homeassistant/components/screenlogic/climate.py

```diff
@@ -37,11 +37,11 @@
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up entry."""
     entities = []
-    data = hass.data[DOMAIN][config_entry.entry_id]
-    coordinator = data["coordinator"]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]["coordinator"]
 
-    for body in data["devices"]["body"]:
+    for body in coordinator.data[SL_DATA.KEY_BODIES]:
```

Comment:
You could do:

```
    async_add_entities(
        [
            ScreenLogicClimate(coordinator, body)
            for body in coordinator.data[SL_DATA.KEY_BODIES]
        ]
    )
```

...but it probably not worth it

Reply:
True. I could do this with the switches, too.

Reply:
Probably not worth it if you end up adding more.